README.md

```diff
@@ -18,3 +18,41 @@ error instead of silently succeeding (which `conda env` would do, thus
 ruining the next 3 weekends because your environment doesnt not contain what
 you specified it to contain)
 
+# Why
+ 
+ TL/DR: conda <4.6 , when calling `conda env` install/force ignores the channel pin. 
+ 
+ We, and many other people use conda environment yamls, such as below, to describe self contained environments
+ using packages from different channels. I'm not sure if the conda developers are deprecating the `conda env` command, but it is common in conda version <4.6 that conda will simply ignore the channel pins `channel::package` and take the most recent build across all channels listed.
+
+```
+name: stitchup
+
+channels:
+  - loopbio
+  - conda-forge
+  - defaults
+
+dependencies:
+  - conda-forge::python=2.7
+  - loopbio::ffmpeg=4.0.2=nonvidia_gpl*
+  - loopbio::opencv=3.4.3
+```
+
+Depending on who (which channel) has the most recent opencv build of that version, a different package will be installed. When combined with an environment of any decent complexty, and with the fact that conda(forge) developers often rebuild minor or micro build increments **with new or changed dependencies!!!** will often lead to environments being installed that do not contain what you specified them to contain.
+
+To make matters worse, `conda env` often **silently succeeds**, which means you will just get random crashes because you have mixed incompatible and/or poorly specified (by other packages) dependencies.
+
+If you want to use conda envs `environment.yaml` then the only way to guarentee that you get the build from the channel you requested is to use hashed build strings. But if you are doing this, you might as well use explicit environment / lists of packages anyway.
+
+A 'known' [workaround](https://github.com/conda/conda/issues/6385#issuecomment-361787360) has been to just call `conda create` **not conda env** with all the packages in the environment yaml. **That's what this tool does**
+
+TL/DR: let's use the (hashed) build string to ensure conda gets our packages until channel pins in environment.yaml files get respected. From there, we should also try to use "locked" (aka "explicit") environments - although that is more of an upcoming conda feature than a reality in the conda 4.5 series.
```

Comment:
I think this is not correct and I will contradict myself. While we use a conda env version that does not respect channel pins, we should simply use condaenv,